irohad/ametsuchi/impl/postgres_wsv_command.cpp

```diff
@@ -23,71 +25,134 @@ namespace iroha {
-          + transaction_.quote(account_id) + ", "
-          + transaction_.quote(role_name) + ");");
+    WsvCommandResult PostgresWsvCommand::insertAccountRole(
+        const std::string &account_id, const std::string &role_name) {
+      auto result =
+          execute_("INSERT INTO account_has_roles(account_id, role_id) VALUES ("
+                   + transaction_.quote(account_id) + ", "
+                   + transaction_.quote(role_name) + ");");
+
+      auto message_gen = [&] {
+        return (boost::format("failed to insert account role, account: '%s', "
+                              "role name: '%s'")
+                % account_id % role_name)
+            .str();
+      };
+
+      return makeCommandResult(result, message_gen);
     }
 
-    bool PostgresWsvCommand::deleteAccountRole(const std::string &account_id,
-                                               const std::string &role_name) {
-      return execute("DELETE FROM account_has_roles WHERE account_id="
-                     + transaction_.quote(account_id)
-                     + "AND role_id=" + transaction_.quote(role_name) + ";");
+    WsvCommandResult PostgresWsvCommand::deleteAccountRole(
+        const std::string &account_id, const std::string &role_name) {
+      auto result = execute_("DELETE FROM account_has_roles WHERE account_id="
+                             + transaction_.quote(account_id) + "AND role_id="
+                             + transaction_.quote(role_name) + ";");
+      auto message_gen = [&] {
+        return (boost::format("failed to delete account role, account id: %s, "
+                              "role name: '%s'")
+                % account_id % role_name)
+            .str();
+      };
+
+      return makeCommandResult(result, message_gen);
     }
 
-    bool PostgresWsvCommand::insertRolePermissions(
+    WsvCommandResult PostgresWsvCommand::insertRolePermissions(
         const std::string &role_id, const std::set<std::string> &permissions) {
       auto entry = [this, &role_id](auto permission) {
         return "(" + transaction_.quote(role_id) + ", "
             + transaction_.quote(permission) + ")";
       };
-      return execute(
+
+      auto result = execute_(
           "INSERT INTO role_has_permissions(role_id, permission_id) VALUES "
           + std::accumulate(
                 std::next(permissions.begin()),
                 permissions.end(),
                 entry(*permissions.begin()),
                 [&entry](auto acc, auto x) { return acc + ", " + entry(x); })
           + ";");
+
+      auto all_permissions =
+          std::accumulate(std::next(permissions.begin()),
+                          permissions.end(),
+                          *permissions.begin(),
+                          [](auto res, auto b) { return res + ", " + b; });
+
+      auto message_gen = [&] {
+        return (boost::format("failed to insert role permissions, role "
+                              "id: '%s', permissions: %s")
```

Comment:
Imo it's better wrap the permissions, maybe like this: `"... permissions: [%s]"`

Reply:
Agreed